### PR TITLE
Fix buildah config --cmd to handle array

### PR DESCRIPTION
**Pull request by rhatdan**
buildah config --cmd '[ "/a", "b". "c" ]' is supposed to add
an array of commands.  This PR fixes this.

This PR also returns errors rather then just logging them so
command will fail with bad data.

Added lots of tests for failure situations, and to make sure
--cmd works correctly.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@TomSweeneyRedHat @vrothberg @giuseppe @nalind PTAL

File: cmd/buildah/config.go [hunk 1]
```diff
@@ -198,24 +216,20 @@ func updateConfig(builder *buildah.Builder, c *cobra.Command, iopts configResult
...
-		cmdSpec, err := shellwords.Parse(cmd)
-		if err != nil {
-			logrus.Errorf("error parsing --cmd %q: %v", cmd, err)
-		} else {
-			builder.SetCmd(cmdSpec)
+		if err := updateCmd(builder, iopts.cmd); err != nil {
+			return err
 		}
-		conditionallyAddHistory(builder, c, "/bin/sh -c #(nop)  CMD %s", cmd)
+		conditionallyAddHistory(builder, c, "/bin/sh -c #(nop)  CMD %s", iopts.cmd)
```
Comment (nalind) [inline cmd/buildah/config.go:232]:
```suggestion
		conditionallyAddHistory(builder, c, "/bin/sh -c #(nop) CMD %s", iopts.cmd)
```

**Comment by nalind** [comment]
Generally LGTM, but you'll probably want to update `updateEntrypoint()` to conform to the error handling strategy in `updateCmd()` and `updateHealthcheck()`.

**Comment by rhatdan** [comment]
updateEntryPoint, does not return any errors, so having it return nil did not make much sense.

**Comment by nalind** [comment]
Ah, right.  LGTM, then.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**